internal/services/dashboard/dashboard_grafana_resource.go

```diff
@@ -192,6 +203,8 @@ func (r DashboardGrafanaResource) Create() sdk.ResourceFunc {
 				deterministicOutboundIP = grafanaresource.DeterministicOutboundIPEnabled
 			}
 
+			grafanaMajorVersion := metadata.ResourceData.Get("grafana_major_version").(string)
```

Comment:
Since this is available on the Model defined above, we can use that rather than pulling this from the config directly:

```suggestion
```